core/src/trezor/wire/thp/context.py

```diff
@@ -0,0 +1,152 @@
+import ustruct
+from micropython import const
+from typing import TYPE_CHECKING
+
+from storage.cache_thp import (
+    BROADCAST_CHANNEL_ID,
+    ChannelCache,
+    iter_allocated_channels,
+)
+from trezor import io, loop
+
+from ..errors import WireBufferError
+from . import (
+    CHANNEL_ALLOCATION_REQ,
+    CODEC_V1,
+    PacketHeader,
+    ThpError,
+    ThpErrorType,
+    channel_manager,
+    checksum,
+    control_byte,
+    get_channel_allocation_response,
+)
+from .channel import Channel
+from .checksum import CHECKSUM_LENGTH
+from .writer import INIT_HEADER_LENGTH, write_payload_to_wire_and_add_checksum
+
+if __debug__:
+    from trezor import log
+
+if TYPE_CHECKING:
+    from trezorio import WireInterface
+
+_CID_REQ_PAYLOAD_LENGTH = const(12)
+
+# Codec_v1 magic constant:
+# "?##" + Failure message type + msg_size + msg_data (code = "Failure_InvalidProtocol")
+CODEC_V1_ERROR_MESSAGE = b"\x3f\x23\x23\x00\x03\x00\x00\x00\x14\x08\x11"
+
+
+class ThpContext:
```

Comment:
Nit: wanna maybe add a sentence or two wrt. what this is and how it is different from the other contexts we have?

Reply:
Perhaps even rename the file to something like `interface_context.py`?